src/compiler/types.ts

```diff
@@ -2916,12 +2916,19 @@ namespace ts {
         visit(node: Node, stop: LintStopMethod, error: LintErrorMethod): void;
     }
 
-    export interface SyntacticLintProviderStatic {
-        new (typescript: typeof ts, args: any): LintWalker;
+    export interface BaseProviderStatic {
+        ["extension-kind"]: ExtensionKind;

```

Comment:
This field should be set-once-and-forget. The small bit of extra effort to write `["extension-kind"]` over `.kind` should hopefully make setting it more meaningful, IMO. (Plus, bare `.kind` is honestly overloaded within the compiler - every class of things has a `.kind`. This kind does not function like those kinds - it is a string, those are enums. Best not confuse them.) I should actually probably also add a `readonly` modifier to it.